### PR TITLE
[AWS][ELB] Split ELB data dashboard into 3 different data dashboard 

### LINKED ISSUE #4800: [AWS] Splitting AWS ELB integration into 3 separate data streams
The current ELB integration, we are collecting metrics from 3 different kinds of load balancers: Classic Load Balancer, Application Load Balancer, and Network Load Balancer. The metric collection for these three load balancers can be found [here](https://github.com/elastic/integrations/blob/main/packages/aws/data_stream/elb_metrics/agent/stream/stream.yml.hbs#L39). They are from 3 different namespaces from CloudWatch. 

This issue is to consider/discuss if we should split elb data stream into 3 different data streams: application_elb, network_elb, and classic_elb. Users will get the flexibility to enable the kind of load balancers they have. Note: Even with the separation, these 3 data streams should still show up under the same ELB integration tile in Kibana.

### LINKED ISSUE #10326: [AWS][ELB] Create a dashboard panel for TargetResponseTime metrics
Create a new dashboard panel for showing the TargetResponseTime metrics . This is applicable only for Application Load Balancer. 
This issue must be taken up after resolving https://github.com/elastic/integrations/issues/4800, to avoid issues such as panels not displaying data when a non-ALB type LB is configured

@agithomas

### PR DESCRIPTION
**Pull request by anil-elastic**
## Proposed commit message
Split ELB data dashboard into 3 different data dashboard 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #4800 
- Closes #10326  

### ALB Metrics

| Source Field | Description |
| --- | --- |
| Consumed LCUs | The number of load balancer capacity units (LCU) used by your load balancer |
| HTTP Code ELB 500 Count | The number of HTTP 500 error codes that originate from the load balancer |
| HTTP Code ELB 502 Count | The number of HTTP 502 error codes that originate from the load balancer |
| HTTP Code ELB 503 Count | The number of HTTP 503 error codes that originate from the load balancer |
| HTTP Code ELB 504 Count | The number of HTTP 504 error codes that originate from the load balancer |
| HTTP Fixed Response Count | The number of fixed-response actions that were successful |
| HTTP Redirect Url Limit Exceeded Count | The number of redirect actions that couldn't be completed because the URL in the response location header is larger than 8K |
| IPv6 Processed Bytes | The total number of bytes processed by the load balancer over IPv6 |
| IPv6 Request Count | The number of IPv6 requests received by the load balancer |
| Processed Bytes | The total number of bytes processed by the load balancer over IPv4 and IPv6 |
| Rejected Connection Count | The number of connections that were rejected because the load balancer had reached its maximum number of connections |
| Rule Evaluations | The number of rules processed by the load balancer given a request rate averaged over an hour |

### NLB Metrics
| Source Field | Description |
| --- | --- |
| Active Flow Count TLS | The total number of concurrent TLS flows (or connections) from clients to targets| 
| Consumed LCUs | The number of load balancer capacity units (LCU) used by your load balancer | 
| New FlowCount TLS | The total number of new TLS flows (or connections) established from clients to targets in the time period | 
| Processed Bytes | The total number of bytes processed by the load balancer, including TCP/IP headers| 
| Processed Bytes TLS | The total number of bytes processed by TLS listeners | 
| TCP Client Reset Count | The total number of reset (RST) packets sent from a client to a target | 
| TCP ELB Reset Count | The total number of reset (RST) packets generated by the load balancer | 
| TCP Target Reset Count | The total number of reset (RST) packets sent from a target to a client | 

### ELB Metrics
| Source Field | Description |
| --- | --- |
| Active Flow Count TLS | The total number of concurrent TLS flows (or connections) from clients to targets| 
| Consumed LCUs | The number of load balancer capacity units (LCU) used by your load balancer | 
| New Flow Count TLS | The total number of new TLS flows (or connections) established from clients to targets in the time period | 
| Processed Bytes | The total number of bytes processed by the load balancer, including TCP/IP headers| 
| Processed Bytes TLS | The total number of bytes processed by TLS listeners | 
| TCP Client Reset Count | The total number of reset (RST) packets sent from a client to a target | 
| TCP ELB Reset Count | The total number of reset (RST) packets generated by the load balancer | 
| TCP Target Reset Count | The total number of reset (RST) packets sent from a target to a client |

### **ELB**

image

### **ALB**

ALB

### **NLB**
NLB

#### **ALB**
metricbeat-aws-alb-overview

#### **ELB**
metricbeat-aws-elb-overview

#### **NLB**
metricbeat-aws-nlb-overview

#### **Existing ELB metric dashboard**
deprecated  metricbeat-aws-elb-overview

#### **AWS-Load-Balancer-Overview**
AWS-Load-Balancer-Overview

### CODE REVIEW

File: packages/aws/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.18.0"
+  changes:
+    - description: Splitting AWS ELB dashboard into 3 separate data dashboard (ELB, ALB and NLB).
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/12345
```
Comment (gpop63) [inline packages/aws/changelog.yml:6]:
```suggestion
    - description: Split the current AWS ELB dashboard into 3 separate dashboards, each focusing on a specific type of load balancer: ELB, ALB, and NLB.
      type: enhancement
      link: https://github.com/elastic/integrations/pull/10444
```

**Review by gpop63 — COMMENTED** [review]
After looking into how to properly display the data across dashboards, I believe we can utilize the `aws.cloudwatch.namespace` field.

We can add a filter in each dashboard (similar to `data_stream.dataset` filter) to ensure `aws.cloudwatch.namespace` corresponds to each specific load balancer type:

- `AWS/NetworkELB` namespace for the NLB dashboard
- `AWS/ApplicationELB` namespace for the ALB dashboard
- `AWS/ELB` namespace for the ELB (classic load balancer) dashboard

We also need to use different breakdown fields for lenses based on the load balancer type:

- `AWS/NetworkELB` and `AWS/ApplicationELB`: breakdown by `aws.dimensions.LoadBalancer`
- `AWS/ELB`: breakdown by `aws.dimensions.LoadBalancerName`

**Comment by agithomas** [comment]
> @agithomas WDYT about this approach?

I think it is a good approach, especially from the performance front.

**Comment by agithomas** [comment]
Anil, below is my quick feedback looking at the attached screenshots

1. Please fix the duplicate `metric` panel title such as `Healthy Host Count`
2. Kindly update the package screenshots as part of this PR
3. Please keep related panels closer. For example, Keep Connection count & New connection count together.
4. There exist panel with duplicate title (Request count). Please see the screenshot attached first.
5. Try to avoid usage of `count` in the panel title, instead make use of Y-axis labels. 
6. You can always rename the control drop. Please refer to control dropdown named `cloud.region`
7. Please fix the label `3xx_count_sum`

**Comment by agithomas** [comment]
@anil-elastic / @gpop63 ,

I think it is important to have an input control to use the values of `aws.dimensions.LoadBalancerName` such that the user can select from multiple Loadbalancers of the same type (ALB / ELB / NLB). 

WDYT?

**Comment by aliabbas-elastic** [comment]
@anil-elastic Please have a look at these review comments

1. For `Healthy/Unhealthy` host count it's showing decimal values. Can we remove them to show it as an integer?
2. For `Http 3xx count` check the X-axis label. It's `@timestamp per minute` whereas rest all are 5 minutes. Generally the convention is to keep horizontal axis timestamp interval to `Auto`. Also please check y axis label it is showing `/s`.
3. Related to point 1, a general comment for all visualizations. It's not always necessary to show metrics as `double` values if they are coming as `integer`. Like for example you can remove decimals for 3xx count, 4xx count visualizations. Please check for similar if the decimals are really required.
4. (Minor) It would be great to update the screenshot of `ELB` as it is distorted.

**Comment by anil-elastic** [comment]
@agithomas @aliabbas-elastic I just pushed the changes and updated the snapshot also. Please review.

**Comment by anil-elastic** [comment]
@tommyers-elastic How we can proceed with the old ELB dashboard once these new 3 dashboard merged and released.

1. Do you want to delete the old dashboard.
2. if you want to retain it then we can add some link to point to new dashboards.

File: packages/aws/kibana/dashboard/aws-34021716-10b1-4338-807c-1a9bcb3da222.json [hunk 2]
```diff
@@ -535,70 +591,42 @@
...
                     "x": 0,
-                    "y": 0
+                    "y": 5
                 },
-                "panelIndex": "5",
-                "title": "Healthy Host Count",
-                "type": "lens",
-                "version": "8.6.2"
+                "panelIndex": "bd9e59ab-cf94-49c3-afe7-beb3bbdb14fb",
+                "title": "Unhealthy Host Count",
```
Comment (mbondyra) [inline packages/aws/kibana/dashboard/aws-34021716-10b1-4338-807c-1a9bcb3da222.json:600]: Nit: for both metric visualizations, "Healthy Host Count" and "Unhealthy Host Count" you could color a background whether depending on the value or statically.

**Review by mbondyra — APPROVED** [review]
Visualizations LGTM 👌🏼

**Comment by aliabbas-elastic** [comment]
> @agithomas @aliabbas-elastic I just pushed the changes and updated the snapshot also. Please review.

@anil-elastic Can you also update the screenshots in the PR description once you have done all the changes? It'll be easy to review the changes suggested.

**Review by aliabbas-elastic — APPROVED** [review]
I didn't just check in the `Updated` ones for the visualizations updated :) Thanks for addressing all the comments!

**Review by efd6 — APPROVED** [review]
Purely due to codeowners.

**Comment by anil-elastic** [comment]
Updated the existing ELB metrics dashboard with links of new LB's dashboard. Snapshot attached.

**Comment by agithomas** [comment]
@anil-elastic , can you revisit the content for `Network Load Balancer (NLB) in the modified (existing) overview dashboard ?

**Comment by agithomas** [comment]
Please update the screenshot of the existing ELB overview dashboard to reflect the new layout.

**Comment by gpop63** [comment]
Overall, it looks good to me. A few nitpicks:

- If we use a static color in the overview dashboard, the background colors of the lenses won't change dynamically based on the Kibana color theme.
- For some lenses, we have Y-axis labels, while for others we don't. The current labels seem ok.
- Since the old dashboard has become an overview dashboard with navigation, can we rename it to something like `[Metrics AWS] Load Balancers Overview`? There are many dashboards where we do this, and it would avoid having to deprecate it?

@agithomas

**Comment by anil-elastic** [comment]
@agithomas Updated as per the review comments. Please review once.

File: packages/aws/manifest.yml [hunk 3]
```diff
@@ -298,9 +298,21 @@ policy_templates:
         size: 32x32
         type: image/svg+xml
     screenshots:
+      - src: /img/AWS-Load-Balancer-Overview.png
```
Comment (agithomas) [inline packages/aws/manifest.yml:301]: Can you rename the filenames to be all lowercase?

Comment (anil-elastic) [reply packages/aws/manifest.yml]: Updated.

**Review by agithomas — APPROVED** [review]
LGTM!

File: packages/aws/kibana/dashboard/aws-34021716-10b1-4338-807c-1a9bcb3da222.json (hunk 2 above)
Comment (anil-elastic) [reply packages/aws/kibana/dashboard/aws-34021716-10b1-4338-807c-1a9bcb3da222.json:677]: Color background feature is not available in current version "2.22.0".

**Pull request merged**